pkg/config/utils.go

```diff
@@ -409,94 +417,151 @@ func checkConfig(atmosConfig schema.AtmosConfiguration, isProcessStack bool) err
 	return nil
 }
 
-func processCommandLineArgs(atmosConfig *schema.AtmosConfiguration, configAndStacksInfo schema.ConfigAndStacksInfo) error {
+const cmdLineArg = "Set using command line argument"
+
+func processCommandLineArgs(atmosConfig *schema.AtmosConfiguration, configAndStacksInfo *schema.ConfigAndStacksInfo) error {
+	if err := setBasePaths(atmosConfig, configAndStacksInfo); err != nil {
+		return err
+	}
+	if err := setTerraformConfig(atmosConfig, configAndStacksInfo); err != nil {
+		return err
+	}
+	if err := setHelmfileConfig(atmosConfig, configAndStacksInfo); err != nil {
+		return err
+	}
+	if err := setStacksConfig(atmosConfig, configAndStacksInfo); err != nil {
+		return err
+	}
+	if err := setFeatureFlags(atmosConfig, configAndStacksInfo); err != nil {
+		return err
+	}
+	if err := setSchemaDirs(atmosConfig, configAndStacksInfo); err != nil {
+		return err
+	}
+	if err := setLoggingConfig(atmosConfig, configAndStacksInfo); err != nil {
+		return err
+	}
+	if err := setSettingsConfig(atmosConfig, configAndStacksInfo); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func setBasePaths(atmosConfig *schema.AtmosConfiguration, configAndStacksInfo *schema.ConfigAndStacksInfo) error {
 	if len(configAndStacksInfo.BasePath) > 0 {
 		atmosConfig.BasePath = configAndStacksInfo.BasePath
-		u.LogDebug(fmt.Sprintf("Using command line argument '%s' as base path for stacks and components", configAndStacksInfo.BasePath))
+		log.Debug(cmdLineArg, "BasePath", configAndStacksInfo.BasePath)
 	}
+	return nil
+}
+
+func setTerraformConfig(atmosConfig *schema.AtmosConfiguration, configAndStacksInfo *schema.ConfigAndStacksInfo) error {
 	if len(configAndStacksInfo.TerraformCommand) > 0 {
 		atmosConfig.Components.Terraform.Command = configAndStacksInfo.TerraformCommand
-		u.LogDebug(fmt.Sprintf("Using command line argument '%s' as terraform executable", configAndStacksInfo.TerraformCommand))
+		log.Debug(cmdLineArg, "terraform.command", configAndStacksInfo.TerraformCommand)
 	}
 	if len(configAndStacksInfo.TerraformDir) > 0 {
 		atmosConfig.Components.Terraform.BasePath = configAndStacksInfo.TerraformDir
-		u.LogDebug(fmt.Sprintf("Using command line argument '%s' as terraform directory", configAndStacksInfo.TerraformDir))
+		log.Debug(cmdLineArg, "terraform.base_path", configAndStacksInfo.TerraformDir)
```

Comment:
How do we know this is `cmdLineArg`?

Reply:
This was old log translated to fix linting. But now that i think about it.
Even though at most places we use the function `ProcessCommandLineArgs` returned object and pass to process `InitCliConfig` unfortunately there is no guarantee that it came from command line arg.

But maybe it would be better if we try to fix this in a separate pr.

Reply:
Then we shouldn't say it came from command line arg :-) can we say something else?

Reply:
We can use something else as log because if we see here:
![image](https://github.com/user-attachments/assets/cf7edf79-0a27-4e58-a832-a7b54a1ff723)
![image](https://github.com/user-attachments/assets/5d76f90d-f931-4e37-985c-3b701cd348d2)

The log statement would be not true.

I see this as an opportunity to pitch more of a cli core strategy that would always use viper and we won't have to handle all these manual arguments that we are handling everywhere.

Let me create a ticket to describe what would help us.

Reply:
Yes, can you create an issue for that